pkg/operator/storageclasscontroller/storageclasscontroller.go

```diff
@@ -145,6 +155,5 @@ func (c *StorageClassController) syncStorageClass(ctx context.Context, policyNam
 	scString = policyReplacer.Replace(scString)
 
 	sc := resourceread.ReadStorageClassV1OrDie([]byte(scString))
-	_, _, err := resourceapply.ApplyStorageClass(ctx, c.kubeClient.StorageV1(), c.recorder, sc)
-	return err
+	return c.scStateEvaluator.EvalAndApplyStorageClass(ctx, sc)
```

Comment:
If a user has opted out of SC creation or set state to unmanaged - should we still sync storage policy and tags? Those operations are expensive and we may as well not do them.

Reply:
Thanks, this is a good point. I added a commit to skip syncing the storage policy if the SC is not managed.

Reply:
Actually I'm going to rework this a bit to avoid `Get`ing the ClusterCSIDriver twice.

Reply:
I pushed a new commit, but it depends on https://github.com/openshift/library-go/pull/1456 merging first and then I will bump library-go in this PR again.